### ISSUE TITLE
Responsive Text Size with Rem

### ISSUE THREAD

**Issue by origami-z** [jpmorganchase/salt-ds#4200]
Currently text size are fixed pixels, which doesn't follow browser text resize.

Any plan with `rem`? 

What's the impact for mobile? e.g. iOS Safari

**Comment by origami-z** [comment]
- +1 from pepper
- +1 from MTK

**Comment by mark-tate** [comment]
@pseys @bhoppers2008 can you comment on this

**Comment by bhoppers2008** [comment]
@pseys @joshwooding are you familiar with the requirement?

**Comment by pseys** [comment]
I haven't looked at this issue before, but I understand the ask. I haven't worked with REMs before (I've worked with px, %, and em but not rem) so there are a few issues I'd want to understand first, for instance: 
- How would we handle this across densities? If `--spacing` and `--size` tokens remained as px values what are the implications?
- Figma doesn't currently support REM, how would we represent the change?
- If a user set a browser default how would this impact density? Would we set a base REM regardless of density and then change each style per density, or set a base per density?
- The + from MTK, is that Dev or Designer led? The latter is usually focused on pixel perfect designs and questions any changes we make, there would be fewer guarantees if we implemented REM and more variability in UX – this isn't a negative in the bigger picture, but I'd want to ensure BLMs are aware of the impact of the change and are onboard with it first.

**Comment by joshwooding** [comment]
Figma is the biggest risk

**Comment by pseys** [comment]
> Figma is the biggest complication. Even though this might be a positive change, is it worth the mismatch

No IMO, although I'd be interested to hear @jake-costa opinion from an accessibility perspective. REM can be set in Figma's Dev Mode but that's a moot point if its being taken away from us internally.

**Comment by jake-costa** [comment]
> > Figma is the biggest complication. Even though this might be a positive change, is it worth the mismatch
> 
> No IMO, although I'd be interested to hear @jake-costa opinion from an accessibility perspective. REM can be set in Figma's Dev Mode but that's a moot point if its being taken away from us internally.

Users must be able to enlarge text using full-page zoom or text-only zoom. (Supporting both techniques is preferable but not required.) Not being able to resize text would be a WCAG 1.4.4 violation for both Web and Mobile.

**Comment by pseys** [comment]
> > > Figma is the biggest complication. Even though this might be a positive change, is it worth the mismatch
> > 
> > 
> > No IMO, although I'd be interested to hear @jake-costa opinion from an accessibility perspective. REM can be set in Figma's Dev Mode but that's a moot point if its being taken away from us internally.
> 
> Users must be able to enlarge text using full-page zoom or text-only zoom. (Supporting both techniques is preferable but not required.) Not being able to resize text would be a WCAG 1.4.4 violation for both Web and Mobile.

Based on this Jake, I assume the recommendation is for us to switch to Rem. I take it this can be limited to just text and not all sizing/spacing. We may have to consider icon sizing alongside this for alignment, but that wouldn't be necessary from an ADA perspective would it?

**Comment by joshwooding** [comment]
> Based on this Jake, I assume the recommendation is for us to switch to Rem. I take it this can be limited to just text and not all sizing/spacing. We may have to consider icon sizing alongside this for alignment, but that wouldn't be necessary from an ADA perspective would it?

There are a lot of systems people use for this, anything you want to scale with text size would use Rems. Increasing the spacing in a box with the text size would re-create our main issue with the current Touch ramp.

**Comment by jake-costa** [comment]
> > > > Figma is the biggest complication. Even though this might be a positive change, is it worth the mismatch
> > > 
> > > 
> > > No IMO, although I'd be interested to hear @jake-costa opinion from an accessibility perspective. REM can be set in Figma's Dev Mode but that's a moot point if its being taken away from us internally.
> > 
> > 
> > Users must be able to enlarge text using full-page zoom or text-only zoom. (Supporting both techniques is preferable but not required.) Not being able to resize text would be a WCAG 1.4.4 violation for both Web and Mobile.
> 
> Based on this Jake, I assume the recommendation is for us to switch to Rem. I take it this can be limited to just text and not all sizing/spacing. We may have to consider icon sizing alongside this for alignment, but that wouldn't be necessary from an ADA perspective would it?

@pseys Text Resize 1.4.4 would not apply to icons though it is still recommended. Also, i don't believe there is a requirement to use Rem, that said, Rem and em are common. Like Josh mentioned, there are a number of ways to handle this, we just don't want to use fixed pixels.

**Comment by mark-tate** [comment]
We should discuss this as part of pokemon
